In [2]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

/kaggle/input/runclustering/RunScores.csv
/kaggle/input/runclustering/consolidated.csv
/kaggle/input/runclustering/RouteUnitCoordinates.csv


Have to figure out the coordinate clusters because I can't download data automatically.  I have to do it manually and using zip codes.

In [9]:
# I need a list of lists, top list is all the coordinates, sublist is ['lat','lon']
# first read the coordinate file
routeCoordDf = pd.read_csv('../input/runclustering/RouteUnitCoordinates.csv')
latlist = routeCoordDf['lat']
lonlist = routeCoordDf['lon']
coordlist = []
for x in range(len(latlist)) :
    coordlist.append([latlist.iloc[x],lonlist.iloc[x]])
print("doneA")

doneA


In [10]:
from sklearn.cluster import KMeans
ckm = KMeans(n_clusters=20)
ckm.fit(coordlist)
ckm.cluster_centers_

array([[ 44.53845502, -93.83408473],
       [ 39.0125231 , -79.95821355],
       [ 30.96771759, -84.4666357 ],
       [ 43.29577165, -71.80487573],
       [ 31.72070082, -90.2132503 ],
       [ 39.22193018, -76.35985149],
       [ 39.76749934, -86.2771057 ],
       [ 35.87956162, -76.99912934],
       [ 38.11810374, -92.30672146],
       [ 34.73908715, -83.79985733],
       [ 45.23027738, -68.96514196],
       [ 47.02496426, -95.2990422 ],
       [ 40.91172204, -74.4773553 ],
       [ 30.60071905, -94.11526452],
       [ 28.08420413, -81.70594462],
       [ 33.11577242, -81.17899961],
       [ 41.49745322, -93.02686429],
       [ 47.33513513, -92.26729451],
       [ 35.46922886, -88.55395709],
       [ 35.63904872, -80.02362788]])

array([[ 44.53845502, -93.83408473],			56011-5400
       [ 39.0125231 , -79.95821355],			26250-7688
       [ 30.96771759, -84.4666357 ],			39834-2324
       [ 43.29577165, -71.80487573],			03278-4518
       [ 31.72070082, -90.2132503 ],			39191-6135
       [ 39.22193018, -76.35985149],			21219-1603
       [ 39.76749934, -86.2771057 ],			46214-3970
       [ 35.87956162, -76.99912934],			27983-8612
       [ 38.11810374, -92.30672146],			65486-A026
       [ 34.73908715, -83.79985733],			30545-2112
       [ 45.23027738, -68.96514196],			04463-1240
       [ 47.02496426, -95.2990422 ],			56470-4181
       [ 40.91172204, -74.4773553 ],			07834-1709
       [ 30.60071905, -94.11526452],			77616-A183
       [ 28.08420413, -81.70594462],			33881-0739
       [ 33.11577242, -81.17899961],			29843-6036
       [ 41.49745322, -93.02686429],			50214-8548
       [ 47.33513513, -92.26729451],			55763-8038
       [ 35.46922886, -88.55395709],			38340-7299
       [ 35.63904872, -80.02362788]])			27239-9334
Results obtained from melissa.com latlngzip4 lookup